plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb.test.ts

```diff
@@ -244,6 +249,85 @@ describe('MongoDBInstrumentation', () => {
     });
   });
 
+  describe('when specifying a responseHook configuration', () => {
+    const dataAttributeName = 'mongodb_data';
+    beforeEach(() => {
+      memoryExporter.reset();
+    });
+
+    describe('with a valid function and enhancedDatabaseReporting set to true', () => {
+      beforeEach(async () => {
+        create({
+          enhancedDatabaseReporting: true,
+          responseHook: (span: Span, result: CommandResult) => {
+            span.setAttribute(dataAttributeName, JSON.stringify(result.result));
+          }
+        });
+      })
+
+      it('should attach response hook data to the resulting span for insert function', done => {
+        const insertData = [{ a: 1 }, { a: 2 }, { a: 3 }];
+        const span = provider.getTracer('default').startSpan('insertRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.insertMany(insertData, (err, result) => {
+            span.end();
+            assert.ifError(err);
+            const spans = memoryExporter.getFinishedSpans();
+            const insertSpan = spans[0];
+
+            assert.strictEqual(insertSpan.attributes[dataAttributeName], JSON.stringify(result.result, Object.keys(result.result).sort()));
+
+            memoryExporter.reset();
+            done();
+          });
+        });
+      });
+
+      it('should attach response hook data to the resulting span for find function', done => {
+        const span = provider.getTracer('default').startSpan('findRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.find({ a: 1 }).toArray((err, results) => {
+            span.end();
+            assert.ifError(err);
+            const spans = memoryExporter.getFinishedSpans();
+            const findSpan = spans[0];
+            const spanResult = JSON.parse(findSpan.attributes[dataAttributeName]?.toString() || '{}');
+
+            assert.strictEqual(spanResult.cursor.firstBatch[0]._id, results[0]._id.toString());
+
+            memoryExporter.reset();
+            done();
+          });
+        });
+      });
+    });
+
+    describe('with an invalid function', () => {
+      beforeEach(async () => {
+        create({
+          enhancedDatabaseReporting: true,
+          responseHook: (span: Span, result: CommandResult) => {
+            throw 'some error';
+          },
+        });
+      });
+
+      it('should not do any harm when throwing an exception', done => {
+        const span = provider.getTracer('default').startSpan('findRootSpan');
+        context.with(trace.setSpan(context.active(), span), () => {
+          collection.find({ a: 1 }).toArray((err, results) => {
+            span.end();
+
+            assert.ifError(err);
```

Comment:
Please assert that the span has indeed been collected (although no data was added to it)

Reply:
Added `assertSpans(spans, 'mongodb.find', SpanKind.CLIENT);`